src/Ether.Vsts/Handlers/Commands/DeleteWorkItemsHandler.cs

```diff
@@ -0,0 +1,47 @@
+﻿using System.Linq;
+using System.Threading.Tasks;
+using Ether.Contracts.Interfaces;
+using Ether.Contracts.Interfaces.CQS;
+using Ether.Vsts.Commands;
+using Ether.Vsts.Dto;
+using static Ether.Contracts.Types.NullUtil;
+
+namespace Ether.Vsts.Handlers.Commands
+{
+    public class DeleteWorkItemsHandler : ICommandHandler<DeleteWorkItems>
+    {
+        private readonly IRepository _repository;
+
+        public DeleteWorkItemsHandler(IRepository repository)
+        {
+            _repository = repository;
+        }
+
+        public async Task Handle(DeleteWorkItems command)
+        {
+            CheckIfArgumentNull(command.Ids, nameof(command.Ids));
+
+            if (!command.Ids.Any())
+            {
+                return;
+            }
+
+            foreach (var id in command.Ids)
```

Comment:
Have you tried `DeleteAsync<WorkItem>(workitem => command.Ids.Contains(workitem.WorkItemId));`?
```suggestion
           await _repository.DeleteAsync<WorkItem>(workitem => command.Ids.Contains(workitem.WorkItemId));
```

Reply:
Changed and tested, it works. Applied in the new comment.